presto-main/src/main/java/io/prestosql/execution/buffer/ClientBuffer.java

```diff
@@ -151,14 +155,18 @@ public void enqueuePages(Collection<SerializedPageReference> pages)
 
     private synchronized void addPages(Collection<SerializedPageReference> pages)
     {
-        pages.forEach(SerializedPageReference::addReference);
+        long rowCount = 0;
+        long bytesAdded = 0;
+        int pageCount = 0;
+        for (SerializedPageReference page : pages) {
+            page.addReference();
+            pageCount++;
+            rowCount += page.getPositionCount();
+            bytesAdded += page.getRetainedSizeInBytes();
+        }
         this.pages.addAll(pages);
-
-        long rowCount = pages.stream().mapToLong(SerializedPageReference::getPositionCount).sum();
         rowsAdded.addAndGet(rowCount);
-        pagesAdded.addAndGet(pages.size());
-
-        long bytesAdded = pages.stream().mapToLong(SerializedPageReference::getRetainedSizeInBytes).sum();
+        pagesAdded.addAndGet(pageCount);
```

Comment:
why manually count pages when you can just use `pages.size()`?

Reply:
Since we're accepting any `Collection<SerializedPageReference>` implemetation and iterating through it already it seemed safer in terms of the performance cliff to count during iteration than to assume a (cheap) O(1) `Collection#size()` implementation.